tests/pytests/unit/modules/test_zpool.py

```diff
@@ -0,0 +1,1111 @@
+    """
+    ret = {}
+    ret["stdout"] = ""
+    ret["stderr"] = ""
+    ret["retcode"] = 0
+    mock_cmd = MagicMock(return_value=ret)
+
+    with patch.dict(zpool.__salt__, {"cmd.run_all": mock_cmd}), patch.dict(
+        zpool.__utils__, utils_patch
+    ):
+        ret = zpool.labelclear("/dev/rdsk/c0t0d0", force=False)
+        res = OrderedDict([("labelcleared", True)])
+        assert ret == res
+
+
+def test_labelclear_nodevice(utils_patch):
+    """
+    Tests labelclear on non existing device
+    """
+    ret = {}
+    ret["stdout"] = ""
+    ret["stderr"] = "failed to open /dev/rdsk/c0t0d0: No such file or directory"
+    ret["retcode"] = 1
+    mock_cmd = MagicMock(return_value=ret)
+
+    with patch.dict(zpool.__salt__, {"cmd.run_all": mock_cmd}), patch.dict(
+        zpool.__utils__, utils_patch
+    ):
+        ret = zpool.labelclear("/dev/rdsk/c0t0d0", force=False)
+        res = OrderedDict(
+            [
+                ("labelcleared", False),
+                (
+                    "error",
+                    "failed to open /dev/rdsk/c0t0d0: No such file or directory",
+                ),
+            ]
+        )
+        assert ret == res
+
+
+def test_labelclear_cleared(utils_patch):
+    """
+    Tests labelclear on device with no label
+    """
+    ret = {}
+    ret["stdout"] = ""
+    ret["stderr"] = "failed to read label from /dev/rdsk/c0t0d0"
+    ret["retcode"] = 1
+    mock_cmd = MagicMock(return_value=ret)
+
+    with patch.dict(zpool.__salt__, {"cmd.run_all": mock_cmd}), patch.dict(
+        zpool.__utils__, utils_patch
+    ):
+        ret = zpool.labelclear("/dev/rdsk/c0t0d0", force=False)
+        res = OrderedDict(
+            [
+                ("labelcleared", False),
+                ("error", "failed to read label from /dev/rdsk/c0t0d0"),
+            ]
+        )
+        assert ret == res
+
+
+def test_labelclear_exported(utils_patch):
+    """
+    Tests labelclear on device with from exported pool
+    """
+    ret = {}
+    ret["stdout"] = ""
+    ret["stderr"] = "\n".join(
+        [
+            "use '-f' to override the following error:",
+            '/dev/rdsk/c0t0d0 is a member of exported pool "mypool"',
+        ]
+    )
+    ret["retcode"] = 1
+    mock_cmd = MagicMock(return_value=ret)
+    with patch.dict(zpool.__salt__, {"cmd.run_all": mock_cmd}), patch.dict(
+        zpool.__utils__, utils_patch
+    ):
+        ret = zpool.labelclear("/dev/rdsk/c0t0d0", force=False)
+        res = OrderedDict(
+            [
+                ("labelcleared", False),
+                (
+                    "error",
+                    "use 'force=True' to override the following"
+                    " error:\n/dev/rdsk/c0t0d0 is a member of exported pool"
+                    ' "mypool"',
+                ),
+            ]
+        )
+        assert ret == res
+
+
+@pytest.mark.skipif(
+    not salt.utils.path.which("mkfile"), reason="Cannot find mkfile executable"
+)
```

Comment:
```suggestion
@pytest.mark.skip_if_binaries_missing("mkfile")
```
And the same approach for the simmiliar skips

Reply:
Fixed